addons/mail/models/mail_thread.py

```diff
@@ -1135,7 +1128,13 @@ def message_new(self, msg_dict, custom_values=None):
         name_field = self._rec_name or 'name'
         if name_field in fields and not data.get('name'):
             data[name_field] = msg_dict.get('subject', '')
-        return self.create(data)
+        try:
+            with self.env.cr.savepoint():
+                record = self.create(data)
+        except exceptions.AccessError:
+            # incase of record rule fails use super user
+            record = self.sudo().create(data)
```

Comment:
since sudo won't change user_id, we could just create it in sudo mode in all cases, no?